tools/deploy_test.py

```diff
@@ -0,0 +1,260 @@
+                seg_pred = torch.nn.functional.interpolate(
+                    seg_pred, size=tuple(ori_shape[:2]), mode='nearest')
+                seg_pred = seg_pred.long().detach().cpu().numpy()
+            seg_pred = seg_pred[0]
+            seg_pred = list(seg_pred)
+            return seg_pred
+
+        def aug_test(self, imgs, img_metas, **kwargs):
+            raise NotImplementedError('This method is not implemented.')
+
+except Exception:
+    pass
+
+# import onnxruntime if needed
+try:
+    from mmcv.tensorrt import TRTWraper, load_tensorrt_plugin
+
+    class TensorRTSegmentor(BaseSegmentor):
+
+        def __init__(self, trt_file: str, cfg: Any, device_id: int):
+            super(TensorRTSegmentor, self).__init__()
+            try:
+                load_tensorrt_plugin()
+            except (ImportError, ModuleNotFoundError):
+                warnings.warn('If input model has custom op from mmcv, \
+                    you may have to build mmcv with TensorRT from source.')
+            model = TRTWraper(
+                trt_file, input_names=['input'], output_names=['output'])
+
+            self.model = model
+            self.device_id = device_id
+            self.cfg = cfg
+            self.test_mode = cfg.model.test_cfg.mode
+
+        def extract_feat(self, imgs):
+            raise NotImplementedError('This method is not implemented.')
+
+        def encode_decode(self, img, img_metas):
+            raise NotImplementedError('This method is not implemented.')
+
+        def forward_train(self, imgs, img_metas, **kwargs):
+            raise NotImplementedError('This method is not implemented.')
+
+        def simple_test(self, img: torch.Tensor, img_meta: Iterable,
+                        **kwargs) -> list:
+            with torch.cuda.device(self.device_id), torch.no_grad():
+                seg_pred = self.model({'input': img})['output']
+            seg_pred = seg_pred.detach().cpu().numpy()
+            # whole might support dynamic reshape
+            ori_shape = img_meta[0]['ori_shape']
+            if not (ori_shape[0] == seg_pred.shape[-2]
+                    and ori_shape[1] == seg_pred.shape[-1]):
+                seg_pred = torch.from_numpy(seg_pred).float()
+                seg_pred = torch.nn.functional.interpolate(
+                    seg_pred, size=tuple(ori_shape[:2]), mode='nearest')
+                seg_pred = seg_pred.long().detach().cpu().numpy()
+            seg_pred = seg_pred[0]
+            seg_pred = list(seg_pred)
+            return seg_pred
+
+        def aug_test(self, imgs, img_metas, **kwargs):
+            raise NotImplementedError('This method is not implemented.')
+
+except Exception:
+    pass
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(
+        description='mmseg backend test (and eval)')
+    parser.add_argument('config', help='test config file path')
+    parser.add_argument('model', help='Input model file')
+    parser.add_argument(
```

Comment:
Can we determine the backend type by the file extension of input model file. For instance, '.onnx' for onnxruntime, and '.trt' for TensorRT?

Reply:
Users might use `.engine`, `.model` or other extensions.